src/main/java/com/coreos/jetcd/ClientBuilder.java

```diff
@@ -165,4 +173,39 @@ public ClientBuilder copy() {
       throw EtcdExceptionFactory.newEtcdException(e);
     }
   }
+
+  private static boolean isValidEndpointFormat(String endpoint) {
+    if (isValidHostAndPort(endpoint)) {
+      return true;
+    }
+
+    URL u = null;
+    try {
+      u = new URL(endpoint);
+    } catch (MalformedURLException e) {
+      return false;
+    }
+
+    if (SCHEMES.stream().noneMatch(u.getProtocol()::equals)) {
+      return false;
+    }
+
+    // endpoint must contain a port.
+    if (u.getPort() == -1) {
+      return false;
+    }
+
+    // endpoint must not contain a path.
+    return u.getPath().isEmpty();
+  }
+
+  private static boolean isValidHostAndPort(String endpoint) {
```

Comment:
it should not. scheme should be specific. unless java client wants to attach http as default scheme.